### DISCUSSION TITLE
Script on 8266: Is there any trigger aviliable to execute some action in the script before deepsleep starts?

### DISCUSSION THREAD

**Discussion by Sfre12** [arendst/Tasmota#18647] [Q&A]
I am using esp8266 with Tasmota and script support.
Now I want to do some tasks which should be executed excactly before deepsleepstarts.
Is there any trigger aviliable?

**Comment by Sfre12** [comment]
Nobody can help with some knowledge about this?

**Reply by gemu2015** [reply]
i don't have any device wired for deep sleep.
however from the code you may try

```
>D
dswake=0
>T
dswake=DeepSleep#Wakeup
if dswake>0 {
print %dswake%
}
```
this should be called just before going to sleep

**Reply by stefanbode** [reply]
@gemu2015 
There is a documentation to use rules. See here: https://tasmota.github.io/docs/DeepSleep/#rules

"System#Save" is called before DeepSleep and can be used as a hook

`Rule1 ON Power1#Boot DO Power on ENDON ON System#Save DO Power off ENDON`

**Reply by Sfre12** [reply]
Thank you. Thats looks like I am searching for. 
As soon as I find the time to test this, I will get back to you.
